akka-contrib/src/main/scala/akka/contrib/pattern/ClusterClient.scala

```diff
@@ -200,7 +200,11 @@ class ClusterClient(
         // heartbeat messages for compatibility with 2.3.12
         receptionist ! Identify(Heartbeat)
       }
-    case ActorIdentity(Heartbeat, _) ⇒
+    case ActorIdentity(Heartbeat, None) ⇒
+      // Server replied, but without ref, i.e. new incarnation (new uid) of receptionist,
+      // which may happen if server is restarted. Don't call failureDetector.
```

Comment:
yes, I was also thinking that since 2.4 does not use this ActorIdentity thingies it could be perceived as a regression (slower behavior) when updating to 2.4 later so I prefer to stick to timeout based failure detection